### PR TITLE
🔀 Merge main to pbft

### PR DESCRIPTION
**Pull request by greymistcube**
Mainly pulling #2321 from the main branch.

### CODE REVIEW

**Review by colibrishin — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -110,6 +110,42 @@ a system action, its features will be added more in the future.
...
+     -  `Block<T>` has constructors `Block<T>(IBlockHeader,
+        IEnumerable<Transaction<T>>)` and `Block<T>(PreEvaluationBlock<T>,
+        (HashDigest<SHA256>, ImmutableArray<byte>, BlockHash))`.
+ -  `BlockContent<T>` no longer inherits `BlockMetadata` and
+    `PreEvaluationBlock<T>` no longer inherits `PreEvaluationBlockHeader`.
+    [[#2231]]
+ -  Both `BlockMetadata` and `BlockContent<T>` are made immutable.  Their
+    properties can no longer be assigned to.  [[#2231]]
+ -  Copy extension methods for `BlockMetadata` and `BlockContent<T>` removed.
+    [[#2231]]
```
Comment (colibrishin) [inline CHANGES.md:148]:
```suggestion
 -  `BlockContent<T>` no longer inherits `BlockMetadata` and
    `PreEvaluationBlock<T>` no longer inherits `PreEvaluationBlockHeader`.
    [[#2231]]
 -  Both `BlockMetadata` and `BlockContent<T>` are made immutable.  Their
    properties can no longer be assigned to.  [[#2321]]
 -  Copy extension methods for `BlockMetadata` and `BlockContent<T>` removed.
    [[#2321]]
```

**Review by colibrishin — CHANGES_REQUESTED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -130,6 +166,14 @@ a system action, its features will be added more in the future.
     [[#2046], [#2229]]
  -  Added `ActionEvaluator<T>.GenerateRandomSeed()` static method.
     [[#2131], [#2236]]
+ -  Each `BlockMetadata`, `PreEvaluationBlockHeader`, and `BlockHeader`
+    can be accessed from any "larger" type object through properties.  [[#2231]]
```
Comment (colibrishin) [inline CHANGES.md:170]:
```suggestion
 -  Each `BlockMetadata`, `PreEvaluationBlockHeader`, and `BlockHeader`
    can be accessed from any "larger" type object through properties.  [[#2321]]
```

File: CHANGES.md (hunk 1 above)
Comment (greymistcube) [reply CHANGES.md:148]: Yup, but this is fixed in main release https://github.com/planetarium/libplanet/pull/2389.
It'll get fixed once we pull 0.43.0.

File: CHANGES.md (hunk 2 above)
Comment (greymistcube) [reply CHANGES.md:170]: Same as above. 😅

File: CHANGES.md (hunk 1 above)
Comment (colibrishin) [reply CHANGES.md:148]: Oh, Okay.

**Review by colibrishin — APPROVED** [review]

**Review by limebell — APPROVED** [review]

**Pull request merged**